core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/ZippedStream.scala

```diff
@@ -0,0 +1,36 @@
+package com.lynbrookrobotics.potassium.streams
+
+class ZippedStream[A, B](aPeriodicity: ExpectedPeriodicity,
+                         bPeriodicity: ExpectedPeriodicity) extends Stream[(A, B)] {
+  override val expectedPeriodicity: ExpectedPeriodicity = (aPeriodicity, bPeriodicity) match {
+    case (Periodic(a), Periodic(b)) =>
+      if (a == b) {
+        Periodic(a)
+      } else {
+        NonPeriodic
```

Comment:
Sometimes, even if you have a new value from the slower `Stream`, you might have to wait for a value from the newer `Stream` if it updated *just* before the slower stream updated. So this would go against matching the periodicity of the slower `Stream`.